---
title: "Catenin Signature and PARP Inhibition in CTRP Cell Lines"
output: html_notebook
---
```{r}
knitr::opts_chunk$set(echo = FALSE)
```

*Before starting, please make a data_in folder and place the following DepMap files there:*
-"CCLE_expression_21Q3.csv"
-"Drug_sensitivity_AUC_(CTD^2).csv
-"sample_info.csv"

Both of these files are available from [this link](https://depmap.org/portal/download/custom/) 

```{r, results= "hide"}

#install BiocManager if needed
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

#install GSVA if necessary
if (!requireNamespace( "GSVA" , quietly = TRUE)) 
    BiocManager::install("GSVA")

library(tidyverse)
library("GSVA")
library(broom)

```


Outline:
  1. Load in the signature infortmation:
        Signatures from MSigDB (G2M, HDR, etc.)
        Catenin signature information created from differential gene expression analysis
  2. Load in CTRP RNA-Seq data
  3. Format and perform GSVA to score each CTRP cell line 
  4. Load in the PARP inhibitor and Drug inhibitor information 
  5. Perform correlation analysis between GSVA signature scores and Drug Inhibitor
  

# 1. Load in the signature information 
```{r}
files <- dir("./Signatures/", full.names = T, pattern = "\\.gmx")
sig.names <- gsub(pattern = "./Signatures/|\\.gmx", replacement = "", x = files)

signature_list <- list()

#for loop reads in all the .gmx files from the signatures folder
for(i in 1:length(files)){
  signature_list[[i]] <- scan(file = files[[i]], what = "character", skip = 2)
   
  assign(sig.names[[i]], value = signature_list[[i]])
}

#read in the CAT signature
CAT.signatures <- read_tsv(file = "Signatures/Cluster_6_HGNC_Reformat.gct", skip = 2)
Stephen_CAT_signature <- CAT.signatures$NAME

#add CAT signature into the signature list
signature_list[[i+1]] <- Stephen_CAT_signature
sig.names[i+1] <- "Stephen_CAT_signature"
names(signature_list) <- sig.names


```


# 2. Load in CCLE RNA-Seq data and CTRP AUC data

```{r}
#### CCLE Data ####
#comes from DepMap, using Q3 release of 2021, but this dataset hasn't been meaningfully updated since then. 

#expression data is log(TPM) with a psuedo-count of 1

CCLE_Expresssion_raw <- read_csv(file = "../drug-gene-correlation/data_in/CCLE_expression_21Q3.csv")

CCLE <- rename(CCLE_Expresssion_raw, depmap_id = `...1`)

names(CCLE) <- gsub(pattern = " \\(.*\\)", replacement = "", x = names(CCLE)) #clean up the names of the Cell lines


#### CTRP Data ####
#this data also comes from DepMap (https://depmap.org/portal/download/custom/)

CTRP_AUCs_raw <- read_csv(file = "../drug-gene-correlation/data_in/Drug_sensitivity_AUC_(CTD^2).csv")

CTRP <- CTRP_AUCs_raw %>% 
  select(-c(3:6))

colnames(CTRP) <- gsub(pattern = " \\(.*", replacement = "", x = names(CTRP)) #clean up drug names as well


```

#   3. Format and perform GSVA to score each CCLE cell line 

(Below information taken from the GSVA vignette found [here](https://bioconductor.org/packages/devel/bioc/vignettes/GSVA/inst/doc/GSVA.html#3_Overview_of_the_GSVA_functionality) )

To use the `gsva()` function from the GSVA package, we need:

1. A normalized gene expression dataset, which can be provided as:
    A matrix of expression values with genes corresponding to rows and samples corresponding to columns

2. A collection of gene sets; which can be provided in one of the following containers 
    A list object where each element corresponds to a gene set defined by a vector of gene identifiers, and the element names correspond to the names of the gene sets.
    A GeneSetCollection object; see package GSEABase.

TL;DR: Get a gene expression matrix (rows = genes, samples = columns), preferably continuous to use the default parameters of GSVA (our data is log(TPM) units so default will work). Get a list of signatures from signatures created previously

```{r}
#get more gene sets

library("msigdbr")

m_t2g <- msigdbr(species = "Homo sapiens", category = "H") %>%
  dplyr::select(gs_name, human_gene_symbol) %>% 
  group_by(gs_name)

hallmark_gene_list <- split(x = m_t2g$human_gene_symbol, f = m_t2g$gs_name)


```


```{r}
### CCLE RNA-Seq data needs to be transposed 

CCLE.matrix <- as.matrix(CCLE[,-1])
rownames(CCLE.matrix) <- CCLE$depmap_id
CCLE.matrix <- t(CCLE.matrix)

CCLE.matrix[1:5, 1:10]
#looks good now


# The signature list is already a list of gene sets, with names. It looks appropriately formatted 
summary(signature_list)

gsva.es <- gsva(expr = CCLE.matrix, gset.idx.list = signature_list)
gsva.hallmark <- gsva(expr = CCLE.matrix, gset.idx.list = hallmark_gene_list)

gsva.es[1:6, 1:10]

gsva_all_sigs <- rbind(gsva.es, gsva.hallmark)

rownames(gsva_all_sigs)[duplicated(rownames(gsva_all_sigs))] <- paste(rownames(gsva_all_sigs)[duplicated(rownames(gsva_all_sigs))], "2", sep = "_")


```

**Fix, technically the gene sets should be all together before running GSVA (apparently) because otherwise they are slightly off looking at the HALLMARK_G2M_CHECKPOINT which is the appears in both sets 

# 4. Format the Drug inhibitor information

Get the meta-data for CTRP drugs (MOA information) and then CCLE meta data (cell line origin)

```{r}

gsva.es <- gsva_all_sigs

#join the gsva table with CCLE metadata
gsva_ctrp <- as_tibble(t(gsva.es), rownames = "DepMap_ID")
  
CCLE_meta_data <- read_csv(file = "../drug-gene-correlation/data_in/sample_info.csv")

gsva_ccl.meta <- CCLE_meta_data %>% 
  select(DepMap_ID, cell_line_name, primary_disease, Subtype, lineage, lineage_subtype, lineage_sub_subtype) %>% 
  right_join(x = ., y = gsva_ctrp)


#make the CTRP AUC dataset long, combine with the drug meta data from Alex's review

CTRP_long <- CTRP %>% pivot_longer(cols = 3:ncol(.), names_to = "Drug", values_to = "AUCs")

CTRP_meta_data <- readxl::read_xlsx(path = "../drug-gene-correlation/data_in/Harmonized_Compound_Data.xlsx")

CTRP_with_meta <- CTRP_meta_data %>% 
  filter(Dataset == "CTRPv2") %>% 
  select(Compound_Name_in_Dataset, Compound_MOA, Compound_Molecular_Targets) %>% 
  left_join(x = CTRP_long, y = ., by = c("Drug" = "Compound_Name_in_Dataset"))


#now combine the GSVA with CCLE metadata with the CTRP with meta data

drug_gsva_final_data <- left_join(CTRP_with_meta, gsva_ccl.meta, by = c("depmap_id" = "DepMap_ID"))


knitr::kable(drug_gsva_final_data[1:10, ])
  


```



#  5. Perform correlation analysis between GSVA signature scores and Drug Inhibitor

```{r}
sig.names <- c(names(signature_list), names(hallmark_gene_list))

#now, for each of the drugs, correlate their AUC value with the GSVA 
drug_gsva_final_data  %>% 
  group_by(Drug) %>% 
  drop_na(AUCs, any_of(sig.names)) %>%
  summarise_at(vars({{sig.names}}), ~cor(AUCs,., method = "spearman")) %>% knitr::kable()
```

The table above gets a correlation between cell line GSVA score in that pathway and drug response to that drug across all cell lines. But the format is a matrix and harder to work with. Keeping it just in case, but below is better: 

Here is another way of doing it to get the p-values as well:

```{r}
drug_gsva_final_data %>% 
  filter(grepl(x = Compound_Molecular_Targets, pattern = "PARP|WNT" )|grepl(x = Compound_MOA, pattern = "PARP|WNT" ))  %>% #filtering down for testing 
  group_by(Drug) %>% 
  drop_na(AUCs, any_of(sig.names)) %>% #for each drug, drops any row (cell line) where the GSVA or AUC is an NA
  mutate(count = n()) %>% #for each Drug, counts how many cell lines were tested
  group_by(Drug, count) %>% #regroups, just to keep these two things together
  summarise_at(vars({{sig.names}}), ~tidy(cor.test(AUCs,., method = "spearman"))) %>% 
  #fancy summarize, does a cor.test between AUC and each column in sig.names 
    #(vars({{}})) allows the vector to be interpretted as a Tidy input
  #tidy allows the output to be a Tibble, 
    #each drug then has a tibble of correlation results for each signature column
  pivot_longer(cols = sig.names, names_to = "Gene_Set", values_to = "nested_tibbles") %>% 
  #make tidy format and then unnest the tibbles 
  unnest(cols = nested_tibbles) %>% 
  select(Drug, Gene_Set, n_cell_lines = count, spearman_R = estimate, p.value) %>% 
  arrange(p.value) %>% 
  knitr::kable()

```



# 6 Turn that mess into a function so we can do this for any drug list or any subset of cell lines


```{r}
#get default cell line and drugs of interest values

drugs_of_interest <- CTRP_with_meta %>% 
  filter(grepl(x = Compound_Molecular_Targets, pattern = "PARP|WNT" )|grepl(x = Compound_MOA, pattern = "PARP|WNT" )) %>% 
  distinct(Drug) %>% 
  pull(Drug)

drugs_of_interest

cell_lines_of_interest <- unique(drug_gsva_final_data$cell_line_name)

```


```{r correlation_function}
#make 2 functions 
#first one just is dynamic to filter for particular drugs, particular signatures, or particular cell lines
#this will also allow for easier graphing of individual pathways

gsva_filtering_function <- function(data_table = drug_gsva_final_data, 
                                    drugs = drugs_of_interest,
                                    cell_lines = cell_lines_of_interest,
                                    signatures = sig.names){
data_table %>% 
  filter(Drug %in% drugs,
         cell_line_name %in% cell_lines) %>% 
  group_by(Drug) %>% 
  drop_na(AUCs, any_of(sig.names)) %>%
  mutate(count = n()) 
}  

#this one will take the output of the filtering function and then get the correlations from before as done in the last code chunk

gsva_correlation_function <- function(gsva_filtering_output = gsva_filtered, signatures = sig.names){
  gsva_filtering_output %>%  
  group_by(Drug, count) %>% 
  summarise_at(vars({{signatures}}), ~tidy(cor.test(AUCs,., method = "spearman"))) %>% 
  pivot_longer(cols = signatures, names_to = "Gene_Set", values_to = "nested_tibbles") %>% 
  unnest(cols = nested_tibbles) %>% 
  select(Drug, Gene_Set, n_cell_lines = count, spearman_R = estimate, p.value) %>% 
  arrange(p.value)
  
}


```


# 7. Make a bunch of graphing function



```{r}

# Make a full table with all the correlations 
# 
# These correlations are done for every CCL, i.e. pan-cancer. That will have to be interrogated further

all_correlations <- gsva_filtering_function(data_table = drug_gsva_final_data, 
                                    drugs = unique(CTRP_with_meta$Drug),
                                    cell_lines = unique(drug_gsva_final_data$cell_line_name),
                                    signatures = sig.names) %>% 
  gsva_correlation_function()


```

Code below makes a graphing function for interrogating correlations for one pathway across all drugs or for one drug across all 56 pathways


```{r}
one_geneset_all_drugs_graph <- function(correlation_output = all_correlations, 
                                        geneset_filter = "HALLMARK_G2M_CHECKPOINT",
                                        drugs_of_interest_color = drugs_of_interest){
  all_correlations %>% 
  filter(Gene_Set == geneset_filter) %>% 
  mutate(PARP_orWNT_inhibitor = if_else(Drug %in% drugs_of_interest_color, "YES", "Other")) %>% 
  ggplot(aes(x = spearman_R, fill = PARP_orWNT_inhibitor)) +
  geom_dotplot(dotsize = 0.65, method = "histodot") +
  scale_y_continuous(NULL, breaks = NULL) + 
  theme_classic() + 
  labs(title = paste("Distribution of Correlations:"),
       subtitle = paste("CTRP cell line drug response correlated with enrichment in\n", geneset_filter, "for all drugs"))
  
}

one_geneset_all_drugs_graph()

one_drug_all_genesets_graph <- function(correlation_output = all_correlations, 
                                        drug_filter = drugs_of_interest[1],
                                        gene_sets_of_interest = sig.names[1:7]){

all_correlations %>% 
  filter(Drug == drug_filter) %>% 
  mutate(Pathways_of_interest = if_else(Gene_Set %in% gene_sets_of_interest, "CAT Associated", "Other")) %>% 
  ggplot(aes(x = spearman_R, fill = Pathways_of_interest)) +
  geom_dotplot(method = "histodot", count = 10) +
  scale_y_continuous(NULL, breaks = NULL) + 
  theme_classic() + 
  labs(title = paste("Distribution of Correlations:"),
       subtitle = paste("For", drug_filter, "Correlation Among All Pathways"))
     }

one_drug_all_genesets_graph()

```



Make a graphing function for interrogating individual drug:pathway relationships

```{r gsva_graphing_function}
gsva_graphing_function <- function(dataset, drug, gene_set, color = "none"){
  
  gene_set_var <- sym(gene_set)
  
  if(color == "none"){
  dataset %>%
  filter(Drug == drug) %>% 
  ggplot(mapping = aes(x = !!gene_set_var, y = AUCs)) + 
  geom_point() + 
  geom_smooth(method = "lm") + 
  ggpubr::stat_cor(method = "spearman") +
    labs(title = paste(drug, gene_set, sep = "::"))
  } else {
  dataset %>%
  filter(Drug == drug) %>% 
  mutate(colored_cell_lines = if_else(cell_line_name %in% color, "T-Cell", "Other")) %>% 
  ggplot(mapping = aes(x = !!gene_set_var, y = AUCs)) + 
  geom_point(mapping = aes(color = colored_cell_lines)) + 
  geom_smooth(method = "lm") + 
  ggpubr::stat_cor(method = "spearman") +
    labs(title = paste(drug, gene_set, sep = "::")) + 
    scale_color_manual(values = c("black", "blue"))
  }
  
}

gsva_filtering_function() %>% 
gsva_graphing_function(drug = "olaparib", gene_set = "HALLMARK_G2M_CHECKPOINT") 


```

# 9. Analysis: All CTRP Cell Lines

```{r}

drugs_of_interest <- CTRP_with_meta %>% 
  filter(grepl(x = Compound_Molecular_Targets, pattern = "PARP|WNT" )|grepl(x = Compound_MOA, pattern = "PARP|WNT" )) %>% 
  distinct(Drug) %>% 
  pull(Drug)

cell_lines_of_interest <- unique(drug_gsva_final_data$cell_line_name)

gsva_filtering_function() %>% 
  gsva_correlation_function()

for(i in 1:length(sig.names)) {
gsva_filtering_function() %>% 
gsva_graphing_function(drug = "olaparib", gene_set = sig.names[i]) %>% 
    print()
}
```



# 10. Leukemia Cell Lines

```{r}

drugs_of_interest <- CTRP_with_meta %>% 
  filter(grepl(x = Compound_Molecular_Targets, pattern = "PARP|WNT" )|grepl(x = Compound_MOA, pattern = "PARP|WNT" )) %>% 
  distinct(Drug) %>% 
  pull(Drug)

cell_lines_of_interest <- drug_gsva_final_data %>% 
  filter(primary_disease == "Leukemia") %>% 
  select(cell_line_name) %>% 
  distinct() %>% 
  pull(cell_line_name)

gsva_filtering_function() %>% 
  gsva_correlation_function()

for(i in 1:length(sig.names)) {
gsva_filtering_function() %>% 
gsva_graphing_function(drug = "olaparib", gene_set = sig.names[i]) %>% 
    print()
}


```



# 11. ALL

```{r}

drugs_of_interest <- CTRP_with_meta %>% 
  filter(grepl(x = Compound_Molecular_Targets, pattern = "PARP|WNT" )|grepl(x = Compound_MOA, pattern = "PARP|WNT" )) %>% 
  distinct(Drug) %>% 
  pull(Drug)

cell_lines_of_interest <- drug_gsva_final_data %>% 
  filter(lineage_subtype == "ALL")  %>% 
  select(cell_line_name) %>% 
  distinct() %>% 
  pull(cell_line_name)


gsva_filtering_function() %>% 
  gsva_correlation_function()

for(i in 1:length(sig.names)) {
gsva_filtering_function() %>% 
gsva_graphing_function(drug = "olaparib", gene_set = sig.names[i]) %>% 
    print()
}
```



# 12. T-ALL

```{r}

drugs_of_interest <- CTRP_with_meta %>% 
  filter(grepl(x = Compound_Molecular_Targets, pattern = "PARP|WNT" )|grepl(x = Compound_MOA, pattern = "PARP|WNT" )) %>% 
  distinct(Drug) %>% 
  pull(Drug)

cell_lines_of_interest <- drug_gsva_final_data %>% 
  filter(lineage_subtype == "ALL") %>% 
  filter(lineage_sub_subtype == "t_cell") %>% 
  select(cell_line_name) %>% 
  distinct() %>% 
  pull(cell_line_name)


gsva_filtering_function() %>% 
  gsva_correlation_function()

for(i in 1:length(sig.names)) {
gsva_filtering_function() %>% 
gsva_graphing_function(drug = "olaparib", gene_set = sig.names[i]) %>% 
    print()
}
```

# 13. Leukemia with T-ALL colored

```{r}
#Stephen said just to use the T-ALL categorization that DepMap uses
# xlsx_stephen_cell_lines <-  readxl::read_xlsx(path = "./GDSC - Cell Lines Metadata_TALL.xlsx")
# stephen_cell_lines <- xlsx_stephen_cell_lines$`Sample Name`
# stephen_cell_lines <- tolower(stephen_cell_lines) 
# stephen_cell_lines <- gsub(pattern = "[[:punct:]]", replacement = "", x = stephen_cell_lines)


drugs_of_interest <- CTRP_with_meta %>% 
  filter(grepl(x = Compound_Molecular_Targets, pattern = "PARP|WNT" )|grepl(x = Compound_MOA, pattern = "PARP|WNT" )) %>% 
  distinct(Drug) %>% 
  pull(Drug)


cell_lines_of_interest <- drug_gsva_final_data %>% 
  filter(primary_disease == "Leukemia") %>% 
  select(cell_line_name) %>% 
  distinct() %>% 
  pull(cell_line_name)


tall_color <- drug_gsva_final_data %>% 
  filter(lineage_subtype == "ALL") %>% 
  filter(lineage_sub_subtype == "t_cell") %>% 
  select(cell_line_name) %>% 
  distinct() %>% 
  pull(cell_line_name)

gsva_filtering_function() %>% 
  gsva_correlation_function() -> gsva_final_correlation_leukemia

gsva_final_correlation_leukemia

for(i in 1:length(sig.names)) {
gsva_filtering_function() %>% 
gsva_graphing_function(drug = "olaparib", gene_set = sig.names[i], color = tall_color) %>% 
    print()
}


```


```{r}
for(i in 1:length(sig.names)) {
  for(j in 1:length(drugs_of_interest)){
gsva_filtering_function() %>% 
gsva_graphing_function(drug = drugs_of_interest[j], gene_set = sig.names[i], color = tall_color) %>% 
    print()
  }
}

```

# CHECK: IS GSVA the same no matter the number of CCLs used
```{r}
#get TALL cell lines (DEPMAP IDs)
TALL_cell_lines <- drug_gsva_final_data %>% 
  filter(lineage_subtype == "ALL") %>% 
  filter(lineage_sub_subtype == "t_cell") %>% 
  select(depmap_id) %>% 
  distinct() %>% 
  pull(depmap_id)

TALL_cell_lines


#Perform GSVA
gsva.tall <- gsva(expr = CCLE.matrix[,TALL_cell_lines], gset.idx.list = signature_list)

#Print Results and compare to original

gsva.es.subtall <- gsva.es[,TALL_cell_lines]


#the two highly correlate
for(i in 1:6){
cor(gsva.es.subtall[i,], gsva.tall[i,]) %>% print()
}
```

The two highly correlate, but looking at the numbers themselves, the absolute values definitely change. Let's try just the leukemia cell lines

```{r}

#get all leukemia cell lines (DEPMAP IDs)
leukemia_cell_lines <- drug_gsva_final_data %>% 
  filter(primary_disease == "Leukemia") %>% 
  select(depmap_id) %>% 
  distinct() %>% 
  pull(depmap_id)

leukemia_cell_lines


#Perform GSVA
gsva.leukemia <- gsva(expr = CCLE.matrix[,leukemia_cell_lines], gset.idx.list = signature_list)



#join the gsva table with CCLE metadata
gsva_leukemia <- as_tibble(t(gsva.leukemia), rownames = "DepMap_ID")
  
gsva_leukemia.meta <- CCLE_meta_data %>% 
  select(DepMap_ID, cell_line_name, primary_disease, Subtype, lineage, lineage_subtype, lineage_sub_subtype) %>% 
  right_join(x = ., y = gsva_leukemia)


#now combine the GSVA with CCLE metadata with the CTRP with meta data

drug_gsva_leukemia <- inner_join(CTRP_with_meta, gsva_leukemia.meta, by = c("depmap_id" = "DepMap_ID"))



```


# 14. Leukemia GSVA and T-ALL Colored 

```{r}
# now go through the graphing function as before
drugs_of_interest <- CTRP_with_meta %>% 
  filter(grepl(x = Compound_Molecular_Targets, pattern = "PARP|WNT" )|grepl(x = Compound_MOA, pattern = "PARP|WNT" )) %>% 
  distinct(Drug) %>% 
  pull(Drug)


cell_lines_of_interest <- drug_gsva_leukemia %>% 
  select(cell_line_name) %>% 
  distinct() %>% 
  pull(cell_line_name)


tall_color <- drug_gsva_leukemia %>% 
  filter(lineage_subtype == "ALL") %>% 
  filter(lineage_sub_subtype == "t_cell") %>% 
  select(cell_line_name) %>% 
  distinct() %>% 
  pull(cell_line_name)

gsva_filtering_function(data_table = drug_gsva_leukemia) %>% 
  gsva_correlation_function() -> gsva_leukemia_correlation_leukemia

for(i in 1:length(sig.names)) {
gsva_filtering_function(data_table = drug_gsva_leukemia) %>% 
gsva_graphing_function(drug = "olaparib", gene_set = sig.names[i], color = tall_color) %>% 
    print()
}
  

```

My conclusion from this is essentially that they are exactly the same, the leukemia only ones are maybe just slightly better (12 pathways in the leukemia only are significant while 10 are significant with all the cell lines)

The only noteable change is that this filters out some genes because they don't change expression in the leukemia cell lines (usually means they are not expressed) 


# Incorporate GDSC Data

```{r}
Harmonized_CCL_data <- readxl::read_excel(path = "Harmonized_CCL_data.xlsx")
GDSC1 <- read_csv(file = "GDSC1_fitted_dose_response_25Feb20.csv")
GDSC2 <- read_csv(file = "GDSC2_fitted_dose_response_25Feb20.csv")


Harmonized_CCL_data %>% 
  select(Cell_Line_Name_In_Dataset, Dataset, Harmonized_Cell_Line_ID) %>% 
  mutate(RRID = gsub(pattern = ".*RRID:|)", replacement = "", x = Harmonized_Cell_Line_ID)) ->
small_harmonized_CCL

GDSC1 %>% select(CELL_LINE_NAME, DRUG_NAME, LN_IC50) %>% 
  left_join(. , small_harmonized_CCL, by = c("CELL_LINE_NAME" = "Cell_Line_Name_In_Dataset")) %>% 
  select(-Dataset) %>% 
  distinct() ->
RRID_GDSC1
```


```{r}
#gsva_ccl.meta should work still
 
#need DEPMAP IDs though
small_CCLE_metadata <- CCLE_meta_data %>% select(DepMap_ID, RRID)

RRID_GDSC1 %>% left_join(., y = small_CCLE_metadata) %>% 
  select(DepMap_ID, DRUG_NAME, LN_IC50) ->
DepMap_GDSC1

GDSC_with_meta <- CTRP_meta_data %>% 
  filter(Dataset == "GDSC1") %>% 
  select(Compound_Name_in_Dataset, Compound_MOA, Compound_Molecular_Targets) %>% 
  left_join(x = DepMap_GDSC1, y = ., by = c("DRUG_NAME" = "Compound_Name_in_Dataset"))


#now combine the GSVA with CCLE metadata with the CTRP with meta data

drug_gsva_final_data <- left_join(GDSC_with_meta, gsva_ccl.meta)

```



Notes:


GDSC data still needs to be used

Nice to see that the T-ALL cell lines do have a higher GSVA score in all the pathways Stephen gave me compared to any other leukemia:

  Do this, plot GSVA for all cell lines by Tissue type!!!!


Wnt vs PARP inhibitors (correlate); response and also GSVA 

Notch mutations, PTEN and AKT alterations (PTEN mutations are more prevalent)
  PTEN mutation might be effectively the same as B-catenin stabilization
  B-catenin mutations are not common in T-ALL 
  (Notch somehow needs B-catenin it appears)
  Indirect mechanisms leading to B-catenin
  
  
  


