---
title: "Catenin Signature and PARP Inhibition in CTRP Cell Lines"
output:
  html_document:
    df_print: paged
---

```{r}

#install BiocManager if needed
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

#install GSVA if necessary
if (!requireNamespace( "GSVA" , quietly = TRUE)) 
    BiocManager::install("GSVA")

library(tidyverse)
library(GSVA)
```


Outline:
  1. Load in the Catenin signature information created from differential gene expression analysis
  2. Load in CTRP RNA-Seq data
  3. Format and perform GSVA to score each CTRP cell line 
  4. Load in the PARP inhibitor and Drug inhibitor information 
  5. Perform correlation analysis between GSVA signature scores and Drug Inhibitor
  

# 1. Load in the Catenin signature information created from differential gene expression analysis
```{r}
files <- dir("./Signatures/", full.names = T, pattern = "\\.gmx")
sig.names <- gsub(pattern = "./Signatures/|\\.gmx", replacement = "", x = files)

signature_list <- list()

for(i in 1:length(files)){
  signature_list[[i]] <- scan(file = files[[i]], what = "character", skip = 2)
    print(signature_list[[i]])
  assign(sig.names[[i]], value = signature_list[[i]])
}
names(signature_list) <- sig.names
CAT.signatures <- read_tsv(file = "Signatures/Cluster_6_HGNC_Reformat.gct", skip = 2)


```


# 2. Load in CCLE RNA-Seq data and CTRP AUC data

```{r}

##CClE Data
#expression data is log(TPM) with a psuedo-count of 1
CCLE_Expresssion_raw <- read_csv(file = "../drug-gene-correlation/data_in/CCLE_expression_21Q3.csv")

CCLE <- rename(CCLE_Expresssion_raw, depmap_id = `...1`)

names(CCLE) <- gsub(pattern = " \\(.*\\)", replacement = "", x = names(CCLE))




### CTRP Data
CTRP_AUCs_raw <- read_csv(file = "../drug-gene-correlation/data_in/Drug_sensitivity_AUC_(CTD^2).csv")

CTRP <- CTRP_AUCs_raw %>% 
  select(-c(3:6))

colnames(CTRP) <- gsub(pattern = " \\(.*", replacement = "", x = names(CTRP))



CCLE[1:10,1:10]
CTRP[1:10, 1:10]


```

#   3. Format and perform GSVA to score each CCLE cell line 

(Below information taken from the GSVA vignette found [here](https://bioconductor.org/packages/devel/bioc/vignettes/GSVA/inst/doc/GSVA.html#3_Overview_of_the_GSVA_functionality) )

To use the `gsva` function from the GSVA package, we need:

1. A normalized gene expression dataset, which can be provided as:
    A matrix of expression values with genes corresponding to rows and samples corresponding to columns

2. A collection of gene sets; which can be provided in one of the following containers 
    A list object where each element corresponds to a gene set defined by a vector of gene identifiers, and the element names correspond to the names of the gene sets.
    A GeneSetCollection object; see package GSEABase.

TL;DR: Get a gene expression matrix (rows = genes, samples = columns), preferably continuous to use the default parameters of GSVA (our data is log(TPM) units so default will work). Get a list of signatures from signatures created previously



```{r}
### CCLE RNA-Seq data needs to be transposed 

CCLE.matrix <- as.matrix(CCLE[,-1])
rownames(CCLE.matrix) <- CCLE$depmap_id
CCLE.matrix <- t(CCLE.matrix)

CCLE.matrix[1:10, 1:10]
#looks good now

```